t/t2232-job-info-security.t

```diff
@@ -279,4 +279,61 @@ test_expect_success 'flux job wait-event guest.exec.eventlog fails via -p (live
 	flux job cancel $jobid
 '
 
+#
+# job info invalid eventlog formatting corner case coverage
+#
+
+# for these tests we need to create a fake job eventlog in the KVS
+
+test_expect_success 'create empty eventlog for job' '
+	jobidhex=`flux job id --to=dothex 123456789` &&
+	flux kvs put "job.${jobidhex}.eventlog"=""
+'
+
+test_expect_success 'flux job info eventlog works (owner)' '
+	jobid=$(submit_job) &&
+	flux job info $jobid eventlog
+'
+
+test_expect_success 'flux job info eventlog fails (user)' '
+	set_userid 9000 &&
+        flux job info 123456789 eventlog 2>&1 | grep "Protocol error" &&
+	unset_userid
+'
+
+test_expect_success 'create eventlog with invalid first field' '
+	jobidhex=`flux job id --to=dothex 123456789`&&
+	flux kvs put "job.${jobidhex}.eventlog"="foobar"
+'
+
+test_expect_success 'flux job info eventlog fails (user)' '
+	set_userid 9000 &&
+        flux job info 123456789 eventlog 2>&1 | grep "Protocol error" &&
+	unset_userid
+'
+
+test_expect_success 'create eventlog without submit context' '
+        submitstr="{\"timestamp\":123.4,\"name\":\"submit\"}" &&
+	jobidhex=`flux job id --to=dothex 123456789` &&
+	echo $submitstr | flux kvs put --raw "job.${jobidhex}.eventlog"=-
+'
+
+test_expect_success 'flux job info eventlog fails (user)' '
+	set_userid 9000 &&
+        flux job info 123456789 eventlog 2>&1 | grep "Protocol error" &&
+	unset_userid
+'
+
+test_expect_success 'create eventlog without submit userid' '
+        submitstr="{\"timestamp\":123.4,\"name\":\"submit\",\"context\":{}}" &&
+	jobidhex=`flux job id --to=dothex 123456789` &&
+	echo $submitstr | flux kvs put --raw "job.${jobidhex}.eventlog"=-
+'
+
+test_expect_success 'flux job info eventlog fails (user)' '
+	set_userid 9000 &&
+        flux job info 123456789 eventlog 2>&1 | grep "Protocol error" &&
+	unset_userid
+'
+
```

Comment:
Also might be a good idea to add tests with bad data (poorly formatted JSON, binary data, etc) here.